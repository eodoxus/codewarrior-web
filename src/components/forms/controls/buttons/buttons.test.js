import React from "react";
import ReactDOM from "react-dom";
import renderer from "react-test-renderer";
import { LinkButton } from "./";

describe("Buttons", () => {
  describe("<LinkButton />", () => {
    let props;
    beforeEach(function() {
      props = {
        icon: "test",
        size: "med",
        target: "_top",
        url: "somewhere.com"
      };
    });

    it("renders without crashing", () => {
      const div = document.createElement("div");
      ReactDOM.render(
        <LinkButton
          icon={props.icon}
          size={props.size}
          target={props.target}
          url={props.url}
        />,
        div
      );
    });

    it("matches expected snapshot", () => {
      const tree = renderer
        .create(
          <LinkButton
            icon={props.icon}
            size={props.size}
            target={props.target}
            url={props.url}
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });

    it(`defaults size and target props to "sm" and "_blank", respectively`, () => {
      const tree = renderer
        .create(<LinkButton icon={props.icon} url={props.url} />)
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
