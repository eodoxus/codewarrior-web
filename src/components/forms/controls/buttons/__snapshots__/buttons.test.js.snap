// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons <LinkButton /> defaults size and target props to "sm" and "_blank", respectively 1`] = `
<a
  className="btn-sm btn btn-default"
  href="somewhere.com"
  rel="noopener noreferrer"
  target="_blank"
>
  <i
    className="fa fa-test"
  />
</a>
`;

exports[`Buttons <LinkButton /> matches expected snapshot 1`] = `
<a
  className="btn-med btn btn-default"
  href="somewhere.com"
  rel="noopener noreferrer"
  target="_top"
>
  <i
    className="fa fa-test"
  />
</a>
`;
